src/Silex/Application.php

```diff
@@ -65,6 +66,15 @@ public function __construct(array $values = array())
         $this['debug'] = false;
         $this['charset'] = 'UTF-8';
         $this['logger'] = null;
+        $this['logger.request'] = function () {
+            $logger = $this['logger'];
+
+            if ($logger instanceof Logger) {
+                $logger = $logger->withName('request');
+            }
+
+            return $logger;
+        };
```

Comment:
I'd move the registration of `'logger.request'` to the `HttpKernelServiceProvider` instead.

All usages of `$app['logger.request']` could be replaced with `$app['logger.request'] ?? $app['logger']`.